google/datalab/ml/_metrics.py

```diff
@@ -0,0 +1,268 @@
+
+    all_df = []
+    for file_name in self._input_csv_files:
+      with _util.open_local_or_gcs(file_name, mode='r') as f:
+        all_df.append(pd.read_csv(f, names=self._headers))
+    df = pd.concat(all_df, ignore_index=True)
+    return df
+
+  def _get_data_from_bigquery(self, queries):
+    """Get data from bigquery table or query."""
+
+    all_df = []
+    for query in queries:
+      all_df.append(query.execute().result().to_dataframe())
+    df = pd.concat(all_df, ignore_index=True)
+    return df
+
+  def accuracy(self):
+    """Get accuracy numbers for each target and overall.
+
+    Returns:
+      A DataFrame with two columns: 'class' and 'accuracy'. It contains the overall
+      accuracy with class being '_all'.
+
+    Raises:
+      Exception if the CSV headers do not include 'target' or 'predicted', or BigQuery
+      does not return 'target' or 'predicted' column.
+    """
+
+    if self._input_csv_files:
+      df = self._get_data_from_csv_files()
+      if 'target' not in df or 'predicted' not in df:
+        raise ValueError('Cannot find "target" or "predicted" column')
+
+      labels = sorted(set(df['target']) | set(df['predicted']))
+      accuracy_results = []
+
+      for label in labels:
+        correct_count = len(df[(df['target'] == df['predicted']) & (df['target'] == label)])
+        total_count = len(df[(df['target'] == label)])
+        accuracy_results.append({
+            'target': label,
+            'accuracy': float(correct_count) / total_count,
+            'count': total_count
+        })
+
+      total_correct_count = len(df[(df['target'] == df['predicted'])])
+      total_accuracy = float(total_correct_count) / len(df)
+      accuracy_results.append({'target': '_all', 'accuracy': total_accuracy, 'count': len(df)})
+      return pd.DataFrame(accuracy_results)
+    elif self._bigquery:
+      query = bq.Query("""
+SELECT
+  target,
+  SUM(CASE WHEN target=predicted THEN 1 ELSE 0 END)/COUNT(*) as accuracy,
+  COUNT(*) as count
+FROM
+  %s
+GROUP BY
+  target""" % self._bigquery)
+      query_all = bq.Query("""
+SELECT
+  "_all" as target,
+  SUM(CASE WHEN target=predicted THEN 1 ELSE 0 END)/COUNT(*) as accuracy,
+  COUNT(*) as count
+FROM
+  %s""" % self._bigquery)
+
+      df = self._get_data_from_bigquery([query, query_all])
+      return df
+
+  def rmse(self):
+    """Get RMSE for regression model evaluation results.
+
+    Returns:
+      the RMSE float number.
+
+    Raises:
+      Exception if the CSV headers do not include 'target' or 'predicted', or BigQuery
+      does not return 'target' or 'predicted' column. Or if target or predicted is not
+      number.
+    """
+
+    if self._input_csv_files:
+      df = self._get_data_from_csv_files()
+      if 'target' not in df or 'predicted' not in df:
+        raise ValueError('Cannot find "target" or "predicted" column')
+
+      df = df[['target', 'predicted']].apply(pd.to_numeric)
+      mse = mean_squared_error(df['target'], df['predicted'])
```

Comment:
Is mse ever negative or a NaN?

Reply:
Tested it. If NaN, sklearn raises understandable error.